README.md

```diff
@@ -1,32 +1,154 @@
+for example:
+
+```java
+public class StringPoolExample {
+    public static void main(String[] args) {
+        String cat1 = "Tom";
+        String cat2 = "Tom";
+        String cat3 = "Garfield";
+    }
+}
+```
+
+When we assign a string literal to a `String`, the Java Virtual Machine (JVM) looks for any other
+`String` that has the same value (i.e. sequence of characters) in a special place in memory called a
+**string pool**. The **string pool** is a part of the heap used to store
+objects that represent string literals. If the JVM does not find another `String`
+object in the string pool with that value, then it will create a new object.
+But if it _does_ find another `String` with that value, Java will return a reference to its memory
+address and not create another object.
+
+![String Pool with Literals](https://curriculum-content.s3.amazonaws.com/6676/java-mod2-strings/string_heap_1.png)
+
+The JVM uses the String pool to avoid creating unnecessary objects
+since each `String` is immutable and it is usually ok to have multiple variables
+referencing the same `String` object.
+
+NOTE: IntelliJ's Java Visualizer displays strings as literal values stored in the variable,
+as if they were primitives:
+
+![intellij string literals](https://curriculum-content.s3.amazonaws.com/6676/java-mod2-strings/intellij_string.png)
+
+However, the browser-based Java Visualizer at [https://pythontutor.com/java.html](https://pythontutor.com/java.html) lets us choose whether to display strings as literals
+or as objects on the heap.  The default is to display them inline (i.e. as string literals),
+but we can change that:
+
+![strings on heap](https://curriculum-content.s3.amazonaws.com/6676/java-mod2-strings/pythontutor_select_heap.png)
+
+Selecting "render all objects on the heap" will make it easier for us to see how
+the JVM avoids creating unnecessary objects for variables that are
+assigned to the same string literal.  Use  [this link](https://pythontutor.com/visualize.html#code=public%20class%20StringPool%20%7B%0A%20%20%20%20public%20static%20void%20main%28String%5B%5D%20args%29%20%7B%0A%20%20%20%20%20%20%20%20String%20cat1%20%3D%20%22Tom%22%3B%0A%20%20%20%20%20%20%20%20String%20cat2%20%3D%20%22Tom%22%3B%0A%20%20%20%20%20%20%20%20String%20cat3%20%3D%20%22Garfield%22%3B%0A%20%20%20%20%7D%0A%7D&cumulative=false&heapPrimitives=true&mode=edit&origin=opt-frontend.js&py=java&rawInputLstJSON=%5B%5D&textReferences=false)
+to step through the sample program and observe what happens when a variable wants to reference
+a string literal that already exists on the heap.
+
+## String Constructor
+
+Now what if we want to have two different `String` objects with the same sequence of characters?
+Java's `String` class actually comes with several constructors, so we can use the `new`
+operator followed by a constructor call such as `String()`
+to help instantiate different `String` objects with the same character sequence!
+
+Recall that a constructor is a special
```

Comment:
I'm not sure constructors have been introduced at this point? So maybe remove the word "Recall"

Reply:
I talk about constructors being a special method for creating objects in the introductory object/class lesson.